### PR TITLE
Bug 1819479: Only create "SuccessfulUpdate" events when updates are made

### PR DESCRIPTION
**Pull request by joelsmith**
Also:
- get release version from deployement object, not VPA controller object
- set pod spec defaults so that DeepEquals returns true for equivalent
  pod spec templates

### CODE REVIEW

**Comment by joelsmith** [comment]
@rphillips PTAL when you have a sec. Thanks!
/assign @rphillips

File: pkg/controller/verticalpodautoscaler/verticalpodautoscaler_controller.go [hunk 1]
```diff
@@ -257,17 +257,17 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 			klog.Info(msg)
 			continue
 		}
-		if err := r.UpdateAutoscaler(vpa, params); err != nil {
+		if err, updated := r.UpdateAutoscaler(vpa, params); err != nil {
```
Comment (rphillips) [inline pkg/controller/verticalpodautoscaler/verticalpodautoscaler_controller.go:260]: let's reverse the order here of the returned error, usually golang does the error last, (updated, err).

Comment (joelsmith) [reply pkg/controller/verticalpodautoscaler/verticalpodautoscaler_controller.go]: Thanks. Updated.

**Pull request merged**